File: GithubDashboard/app/src/main/java/com/zimpo/githubdashboard/Activity/MainActivity.java
package com.zimpo.githubdashboard.Activity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.zimpo.githubdashboard.Api.GithubUserApi;
import com.zimpo.githubdashboard.Object.User;
import com.zimpo.githubdashboard.R;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        configure();
    }

    private EditText searchET = null;
    private TextView warningTV = null;
    private Button searchB = null;
    private Button getB = null;

    public void configure()
    {
        //VARIABLES
        searchET = (EditText) findViewById(R.id.searchET);
        warningTV = (TextView) findViewById(R.id.warningTV);
        searchB = (Button) findViewById(R.id.searchB);
        getB = (Button) findViewById(R.id.getB);

        searchUser();

        getUser();
    }

    public void searchUser()
    {

    }

    public void getUser()
    {
        getB.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
               if (searchET.getText().toString().length() > 0)
               {
                   warningTV.setVisibility(View.GONE);

                   GithubUserApi githubUserApi = new GithubUserApi(getApplicationContext(), new GithubUserApi.ProcessFinish() {
                       @Override
                       public void processFinish(ArrayList<User> users) {
                           if (users != null && users.size() > 0)
                           {
                               Intent intent = new Intent(getApplicationContext(), UserActivity.class);
                               intent.putExtra("EXTRA_USER", users.get(0));
                               startActivity(intent);
                           }
                           else
                               Toast.makeText(getApplicationContext(), "Oups, un problème est survenu", Toast.LENGTH_LONG).show();
                       }
                   });
                   githubUserApi.execute(searchET.getText().toString());

               }
               else
                   warningTV.setVisibility(View.VISIBLE);
            }
        });
    }
}

File: GithubDashboard/app/src/main/java/com/zimpo/githubdashboard/Activity/MainActivity.kt
package com.zimpo.githubdashboard.Activity

import androidx.appcompat.app.AppCompatActivity

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast

import com.zimpo.githubdashboard.Api.GithubUserApi
import com.zimpo.githubdashboard.R


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        configure()
    }

    private lateinit var searchET: EditText
    private lateinit var warningTV: TextView
    private lateinit var searchB: Button
    private lateinit var getB: Button

    fun configure() {
        //VARIABLES
        searchET = findViewById(R.id.searchET) as EditText
        warningTV = findViewById(R.id.warningTV) as TextView
        searchB = findViewById(R.id.searchB) as Button
        getB = findViewById(R.id.getB) as Button

        searchUser()

        getUser()
    }

    fun searchUser() {
        searchB.setOnClickListener {
            if (searchET.text.toString().length > 0) {
                warningTV.visibility = View.GONE

                val intent = Intent(applicationContext, UserListActivity::class.java)
                intent.putExtra("EXTRA_SEARCH", searchET.text.toString())
                startActivity(intent)

            } else
                warningTV.visibility = View.VISIBLE
        }
    }

    fun getUser() {
        getB.setOnClickListener {
            if (searchET.text.toString().length > 0) {
                warningTV.visibility = View.GONE

                val githubUserApi = GithubUserApi(applicationContext, GithubUserApi.ProcessFinish { users ->

                    if (users != null && users.size > 0) {
                        val intent = Intent(applicationContext, UserActivity::class.java)
                        intent.putExtra("EXTRA_USER", users[0])
                        startActivity(intent)
                    } else
                        Toast.makeText(applicationContext, "Oups, un problème est survenu", Toast.LENGTH_LONG).show()

                })
                githubUserApi.execute(searchET.text.toString())

            } else
                warningTV.visibility = View.VISIBLE
        }
    }
}